README.rst

```diff
@@ -0,0 +1,259 @@
+===========
+Babymailgun
+===========
+
+This is my dev test submission for Mailgun.
+
+The project is implemented using a combination of REST API and asynchronous worker architecture all backed by MongoDB.
+The API allows both sending of emails as well fetching the status of emails that have been created.
+Additionally provided is a CLI tool to make interaction with the API easier.
+
+==========
+Jumping In
+==========
+
+You'll need `Docker CE <https://docs.docker.com/engine/installation/>`_, `Docker Compose <https://docs.docker.com/compose/install/>`_ and make
+
+A Makefile is included in the project root to help streamline playing with the project, and has it's own help text to faciliate navigation.
+
+.. code-block:: bash
+
+    make help
+
+Clone the project:
+
+.. code-block:: bash
+
+    git clone https://github.com/Cerberus98/babymailgun.git
+
+To kick off the compose environment, simply run the following from the root of your clone
+
+.. code-block:: bash
+
+    make run && make logs
+
+Docker-compose will build two containers and fetch two additional ones which may take a while depending on your connection. Once it's done, you
+should have a running environment.
+
+To stop all of the running containers, CTRL+C out of the running log output and then:
+
+.. code-block:: bash
+
+    make stop
+
+===============
+Running the CLI
+===============
+
+There are two ways to setup and use the CLI
+
+The Easy Way
+============
+
+I've created a separate Makefile task just for running the CLI against Python 3.6 which makes it pretty easy to jump in and play around
+
+From the project root, first make sure the Docker Compose environment is running as above. Then:
+
+.. code-block:: bash
+
+    make shell
+
+The Hard Way
+============
+
+If you have pyenv installed, install and switch to Python 3.6.0 or better.
+
+For example:
+
+.. code-block:: bash
+
+    pyenv install 3.6.0
+
+Then:
+
+.. code-block:: bash
+
+    pyenv local 3.6.0
+
+Once Python 3.6+ is installed, you'll need to install the project and dependencies. I highly recommend a virtualenv, and recommend Virtualenvwrapper_ doubly so.
+ 
+.. _Virtualenvwrapper: https://virtualenvwrapper.readthedocs.io/en/latest/
+
+The instructions that follow use virtualenvwrapper.
+
+.. code-block:: bash
+
+    ~> mkvirtualenv mailgun
+    # stuff happens
+    ~> cd <project root>
+    ~> make install_python
```

Comment:
👍 for `pyenv`, but I prefer `pyenv-virtualenv`.  Unfortunately the hard way didn't work for me out of the box.  I got an error about `tox` not being present.

Reply:
Hmm, I'll take a look

Reply:
Whoops, fixed: https://github.com/Cerberus98/babymailgun/pull/1/commits/f227fd099443995c237eb7ed226a6b213bef5d20#diff-b67911656ef5d18c4ae36cb6741b7965R16